src/Donors/Routes/RegisterDonorRoutes.php

```diff
@@ -0,0 +1,134 @@
+    {
+        $this->registerGetDonor();
+        $this->registerGetDonors();
+    }
+
+    /**
+     * Get Donor route
+     *
+     * @unreleased
+     */
+    public function registerGetDonor()
+    {
+        register_rest_route(
+            DonorRoute::NAMESPACE,
+            DonorRoute::DONOR,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->donorRequestController->getDonor($request);
+                    },
+                    'permission_callback' => '__return_true',
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * Get Donors route
+     *
+     * @unreleased
+     */
+    public function registerGetDonors()
+    {
+        register_rest_route(
+            DonorRoute::NAMESPACE,
+            DonorRoute::DONORS,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->donorRequestController->getDonors($request);
+                    },
+                    'permission_callback' => '__return_true',
+                ],
+                'args' => [
+                    'page' => [
+                        'type' => 'integer',
+                        'default' => 1,
+                        'minimum' => 1,
+                    ],
+                    'per_page' => [
+                        'type' => 'integer',
+                        'default' => 30,
+                        'minimum' => 1,
+                        'maximum' => 100,
+                    ],
+                    'sort' => [
+                        'type' => 'string',
+                        'default' => 'id',
+                        'enum' => [
+                            'id',
+                            'createdAt',
+                            'name',
+                            'firstName',
+                            'lastName',
+                            'totalAmountDonated',
+                            'totalNumberOfDonations',
+                        ],
+                    ],
+                    'direction' => [
+                        'type' => 'string',
+                        'default' => 'DESC',
+                        'enum' => ['ASC', 'DESC'],
+                    ],
+                    'mode' => [
+                        'type' => 'string',
+                        'default' => 'live',
+                        'enum' => ['live', 'test'],
+                    ],
+                    'onlyWithDonations' => [
+                        'type' => 'boolean',
+                        'default' => true,
+                    ],
```

Comment:
@glaubersilva You alluded to this parameter. This weirds me out. This would make sense if this was a `constituents` endpoint, but having a "I want donors that made donations" is weird. It just naturally raises the question: "How is a person a donor if they didn't donate?" 😆

Reply:
Is there a specific use-case for this filter?

Reply:
@JasonTheAdams In my local dev site, it happens all the time when I delete all donations using the list table bulk actions but don't do the same for donors - check the attached screenshots for reference. 

I believe that this scenario it's an edge case for real users, but it can happen.

So if we need for some reason to return all donors to run a specific action or routine regardless of it they have or not donations, we can set this param to `false`. 

This was my logic to implement it here. Thoughts? 😊

![image](https://github.com/user-attachments/assets/eb0525dd-6b57-4740-a162-59a9e7113bce)

![image](https://github.com/user-attachments/assets/a622e5a3-03be-4821-b9b3-258d67ac6f08)

Reply:
That is a valid reason I have not thought about as a donor's data could technically be deleted.  It's an edge case for sure but possible lol